### ISSUE TITLE
GGMAX - Loading Environment Request

### ISSUE THREAD

**Issue by Necrym59** [TheGameCreators/GameGuruRepo#4154]
AT the moment GGMAX loads to a full screen like this
![current](https://user-images.githubusercontent.com/95015657/217092536-b0d4f2a0-ccde-4af7-ba18-93dbdcdf6368.png)
Using the full screen dimensions, apart from being not very functional in this state it also looks  ameturish  for a quality piece of software, can we please get the hub section to load like this 
![required](https://user-images.githubusercontent.com/95015657/217092888-7469d8a1-22af-46e7-91e0-05f2fba94c6b.png)
and expand out to full screen when either a demo is run or new project is created.

**Comment by MonkeyFrogStudio** [comment]
I really like the hub in the second screen shot. I think that looks neat and professional, imo.

**Comment by synchromesh62** [comment]
It looks better but its not going to look very seamless jumping back and forth from a window to full screen when you selects a game to edit or even go to the hub from the editor and vice versa ?

**Comment by MonkeyFrogStudio** [comment]
I think, once in the editor, if you need to access the HUB, it would just appear on top of the editor. But, in reality, how often does someone bounce back and forth between the editor and the HUB? I know that once I'm working in the editor, I'm usually there for quite a long time. I only use the HUD to get to my game project.

**Comment by synchromesh62** [comment]
Hmm if im honest i kind of like the full hub project screen.
Not sure im on board with this one 

![HUB](https://user-images.githubusercontent.com/35708217/217336785-35ea464e-ddf9-4704-83ba-5ad7128be11f.png)

**Comment by Necrym59** [comment]
Every other software that has preloader/hub windows before launching the actual app dont go into weird full screen modes that doesnt fit the contents, especially on different size monitors. This is why windows was born in the first place. If you want it full screen then fine, just click the fullscreen icon and let it save its size/position like avery other app. Personally i would like the editor itself to save its window size/position as well to be consitant so i can bounce and launch between development apps and folders all on the one screen this is why i bought a large screen in the first place so i could have the spare realestate to work on in multiple apps, like so...
![screen](https://user-images.githubusercontent.com/95015657/217348924-ec33a60a-a9b9-4e95-bef7-bbbdea0563c5.png)
..instead the hub and editor always load full screen and dont remember their window size/position, this needs to be fixed.

**Comment by synchromesh62** [comment]
Yes on an ultrawide Monitor i can imagine it does not look good.
My pic above is pretty much exactly how it looks to me on my standard 28 inch.
So in that sense i see your point.

However on a personal preference note. 
If that is what you guys see just for Windows its certainly enough to put me off buying one.
Looks like a letterbox to me :)

**Comment by Necrym59** [comment]
Also while we are at it can we get this absurdness with the header bar fixed
On full screen the headerbar does not need to be resized in height it nearly takes up half the screen
![story_fullscreen](https://user-images.githubusercontent.com/95015657/217461896-79ef58e7-314b-40ea-a125-c0a5255e32fb.png)
I actually fit more on the screen when i shrink the window - which once again size/position isnt retained. anyway.
![story_windowed](https://user-images.githubusercontent.com/95015657/217462091-d7f76275-8112-4cfd-8f74-9dbed5349f36.png)

**Comment by MonkeyFrogStudio** [comment]
@Necrym59 I'd requested the storyboard header to be reduced here:

https://github.com/TheGameCreators/GameGuruRepo/issues/4122

**Comment by davetgc** [comment]
Are there any suggestions for this, maybe before we look at changing it, we could see some more examples.

**Comment by MonkeyFrogStudio** [comment]
The idea is pretty simple and pretty common. Instead of MAX staring with a full-screen HUD presenting the end-user with options (demos, their game projects, etc.), it opens a WINDOW that is not full screen with nice, appropriately sized icons (not these HUGE icons that, on my screen, are larger than a playing card each). Once you make your selection, it opens the MAX Storyboard to full-screen, indicating you're in MAX-proper (so to speak). If you ever want to access the HUD again, there would be a menu link or button for that. Clicking it would bring up the HUD window on top, allowing the end-user to select the next thing to load. For those that want the initial HUD to be full-screen, they would only have to click the appropriate Window icon to expand it to full-screen. Done.

Here's an example from another game engine called Flax:

![projects-page](https://user-images.githubusercontent.com/36735507/217525012-1555734a-f257-4ad6-82d8-cf0e0cf9fdfc.jpg)

It opens as a smaller window in the middle of your monitor. You can resize it as you please and the next time you open it, it will remember that size.

**Comment by Necrym59** [comment]
Without major work a simple ini entry for window size/position would suffice to start with for the first two windows so it remembers its size/position etc. example:- in a file called GameGuruMax.ini
    [HubWindow]
    Maximize=No
    Minimize=No
    XPos=123
    YPos=48
    Width=248
    Height=198
    [GameGuruEditor]
    Maximize=No
    Minimize=No
    XPos=100
    YPos=134
    Width=1600
    Height=900
![BothWindows](https://user-images.githubusercontent.com/95015657/217642562-04618c92-8eeb-40cc-afc8-06efc1cb5630.png)

But the iu also needs work internally for the sizing of objects like the headerbar, it looks like whoever did this ui was a web designer.  Image scaling should be a simple afair based on width not height in 98% of cases in ui systems unless its specifically required.
Instead of this:-
![IncorrectHeaderBar](https://user-images.githubusercontent.com/95015657/217646026-da44a78a-aecd-4e28-88b8-548a5d80d82a.png)
Somthing like this would be more productive and presentable for example
![CorrectHeaderBar](https://user-images.githubusercontent.com/95015657/217646236-6aa40ffb-4466-4365-8ea5-9af556a27c0b.png)

**Comment by MonkeyFrogStudio** [comment]
The storyboard header should be even thinner, imo. There's no reason to take up that much space to remind me that I'm in the storyboard editor. Cutting even your last one in half would free up room for one more row of nodes! ;)

**Comment by Necrym59** [comment]
I agree thyats just an example it should be hight independent
![Header independent](https://user-images.githubusercontent.com/95015657/217647194-802270de-b1fe-49af-84f5-336c8391ab0d.png)

**Comment by Mt-Chiliad-UFO** [comment]
I think those hubs are cluttered and a bad start. They should be the last thing a developer sees, not the first.
An easy fix is obvious.
I agree that the hub feature opening at the start is an unprofessional (and unpleasant) style. An empty level editing map should precede the hub screen, and not vise-versa. The hub opening up at the start exists only to engineer a full fledged project, or to showcase demos that a user may never want anything to do with. (early days MAX centered promotion for youtubers)- and not pretty.
The easy fix; Can we please just get MAX to open to an empty (full screen) map editor straight from the desktop all the time!
Two options after that;
A> The hub is hovering over this empty level editor-- ready to work in, or to click off and your ready to create on the level.
B> There is NO HUB popup at the start at all. Instead when your project is getting matured, you could easily find that obvious hub button that opens up a tidy hub box, (as we are discussing above) with another quick button taking HUB to full screen.
This would put MAX better directed to feel professional like other, popular and established engines.

**Comment by Necrym59** [comment]
This one shouldnt be waiting on information it needs action it been called out before and should be passed on to whever is doing the UI.

**Comment by MonkeyFrogStudio** [comment]
They do that a lot - mark something as "Waiting More Information" and then just leave it like that. It's like these rarely get revisited or something ...

**Comment by synchromesh62** [comment]
Dave could be waiting on more information from Lee if he has contacted him for more info .
He did not actually ask you for any more info.

**Comment by davetgc** [comment]
Updated this now. I wasn't ignoring it, I just wanted to get as much feedback as possible.

**Comment by Necrym59** [comment]
This is important is not an enhancment it effects the basic functionality and presentability of the product.
At the moment on wide screens or ultra wide screens you do not get shown certain options, I know Lee has done some things with the game engine itself but the ui is letting it down. For example the first image below is of the default full screen of the hub - things are missing and not available eg: In the Right hand side pane: NO Play Game Button and NO Edit Game Button for starters, NO Game Title and NO Information Panel AND worst of all NO scroll bar. All you have is the CreateGame Button and the bypass HUB checkbox.
![full_screen](https://user-images.githubusercontent.com/95015657/230256833-7c5af178-7193-4e16-a39e-8654d1d2bb94.png)

If its sized down width wise into a window on the desktop then all the missing is available
![sized_window](https://user-images.githubusercontent.com/95015657/230256875-2c8a0bce-875a-4348-af2e-642a051a0d21.png)

One other problem is that the Title name of the game/demo on the right hand side pain is in black and can hardfly be seen annd should be made white or a colour at least.

**Comment by davetgc** [comment]
I've moved this to high priority, so it's clearly an issue for a number of users.
